internal/testutil/kube_server_compatibility.go

```diff
@@ -44,3 +49,50 @@ func KubeServerMinorVersionInBetweenInclusive(t *testing.T, discoveryClient disc
 
 	return minor >= min && minor <= max
 }
+
+func ConvertMap[T1, T2 comparable, U1, U2 any](m1 map[T1]U1, fT func(T1) T2, fU func(U1) U2) map[T2]U2 {
+	m2 := make(map[T2]U2)
+	for k1, v1 := range m1 {
+		m2[fT(k1)] = fU(v1)
+	}
+	return m2
+}
+
+func CheckServiceAccountExtraFieldsAccountingForChangesInK8s1_30[T ~map[string]V, V ~[]string](
+	t *testing.T,
+	discoveryClient discovery.DiscoveryInterface,
+	extraGeneric T,
+	pod *v1.Pod,
```

Comment:
Maybe parameter names like `actualExtras` and `expectedPodValues` to clarify which is the actual and which is the expected values?

Reply:
Good idea. Will make that change.